e2e/testcases/override_log_level_test.go

```diff
@@ -18,63 +18,88 @@ import (
 	"strings"
 	"testing"
 
-	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"kpt.dev/configsync/e2e/nomostest"
 	"kpt.dev/configsync/e2e/nomostest/ntopts"
+	"kpt.dev/configsync/e2e/nomostest/taskgroup"
 	nomostesting "kpt.dev/configsync/e2e/nomostest/testing"
 	"kpt.dev/configsync/e2e/nomostest/testpredicates"
 	"kpt.dev/configsync/pkg/api/configsync"
 	"kpt.dev/configsync/pkg/api/configsync/v1beta1"
 	"kpt.dev/configsync/pkg/core"
 	"kpt.dev/configsync/pkg/kinds"
-	"kpt.dev/configsync/pkg/metrics"
+	"kpt.dev/configsync/pkg/metadata"
 	"kpt.dev/configsync/pkg/reconcilermanager"
 	"kpt.dev/configsync/pkg/testing/fake"
 )
 
 func TestOverrideRootSyncLogLevel(t *testing.T) {
-	nt := nomostest.New(t, nomostesting.OverrideAPI)
+	nt := nomostest.New(t, nomostesting.OverrideAPI, ntopts.Unstructured)
 
 	rootSyncName := nomostest.RootSyncNN(configsync.RootSyncName)
 	rootReconcilerName := core.RootReconcilerObjectKey(rootSyncName.Name)
 	rootSyncV1 := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
 
+	// add kustomize to enable hydration controller container in root-sync
+	nt.T.Log("Add the kustomize components root directory")
+	nt.Must(nt.RootRepos[configsync.RootSyncName].Copy("../testdata/hydration/kustomize-components", "."))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("add DRY configs to the repository"))
+
+	nt.T.Log("Update RootSync to sync from the kustomize-components directory")
+	nt.MustMergePatch(rootSyncV1, `{"spec": {"git": {"dir": "kustomize-components"}}}`)
+	syncDirMap := map[types.NamespacedName]string{
+		nomostest.DefaultRootRepoNamespacedName: "kustomize-components",
+	}
+
+	if err := nt.WatchForAllSyncs(nomostest.WithSyncDirectoryMap(syncDirMap)); err != nil {
+		nt.T.Fatal(err)
+	}
+
 	// validate initial container log level value
 	err := nt.Watcher.WatchObject(kinds.Deployment(),
 		rootReconcilerName.Name, rootReconcilerName.Namespace,
 		[]testpredicates.Predicate{
 			testpredicates.DeploymentContainerArgsContains(reconcilermanager.Reconciler, "-v=0"),
 			testpredicates.DeploymentContainerArgsContains(reconcilermanager.GitSync, "-v=5"),
-			testpredicates.DeploymentContainerArgsContains(metrics.OtelAgentName, "-v=0"),
```

Comment:
https://github.com/GoogleContainerTools/kpt-config-sync/pull/1067 should fix the issue with otel-agent logging, so we shouldn't need to remove that part of the assertions any more. I think the addition of hydration-controller validation is still useful though if you want to rebase this change and update the commit message and PR description

Reply:
done. Updated PR to focus on adding `hydration-controller` to test.